hs/app/reachpc/ReachPC/Config.hs

```diff
@@ -42,113 +44,207 @@ readEnum :: forall a. (Show a, Read a, Enum a, Bounded a) => String -> String ->
-  <*> T.dioptional (T.string "project") T..= rtml_project
-
--- TODO: put into separate file and include at compile time
-defaultReachToml :: ByteString
-defaultReachToml = $(makeRelativeToProject "app/reachpc/ReachPC/default_reach.toml" >>= embedFile)
+    if exists then return $ Just path else go paths
+
+globalsTomlPath :: IO FilePath
+globalsTomlPath = getXdgDirectory XdgConfig "reach" <&> (</> "global.toml")
+
+-- helper for interactiveCreate*
+data PromptAnswer = Unset | Unchanged | Str String
+
+parseAnswer :: (Maybe a) -> (String -> a) -> PromptAnswer -> Maybe a
+parseAnswer unchanged readFn = \case
+  Unset -> Nothing
+  Unchanged -> unchanged
+  Str s -> Just $ readFn s
+
+interactiveCreateReachToml :: Bool -> IO ()
+interactiveCreateReachToml alreadyExists = do
+  oldToml <- if alreadyExists then Just <$> readOrErrorReachToml else return Nothing
+  let (oldTomlPath, oldToml') = fromJust oldToml
+  let showOld field = case alreadyExists of
+        True -> " Currently " <> maybe "unset" (\f -> "\"" <> show f <> "\"") (field oldToml')
+        False -> ""
+  let parseAnswer' field = parseAnswer $ field oldToml'
+
+  -- TODO make a read function for splitting project into organization and project name
+  let projectPrompt = ("\nWhat is the remote name of the project?" <> showOld rtml_project)
+  project <- parseAnswer' rtml_project id <$> askUserString alreadyExists False projectPrompt "Project name: "
+
+  let cloudOrLocalPrompt = ("\nDo you want to use Reach Cloud or Reach Local?" <> showOld rtml_cloudOrLocal)
+  cloudOrLocal <- parseAnswer' rtml_cloudOrLocal readCloudOrLocal <$> askUserEnum' alreadyExists True cloudOrLocalPrompt (showEnum @CloudOrLocal)
+
+  let connectorPrompt = ("\nWhat connector do you want to use?" <> showOld rtml_connector)
+  connector <- parseAnswer' rtml_connector readConnector <$> askUserEnum' alreadyExists True connectorPrompt (showEnum @Connector)
+
+  let forwardEnvVarsPrompt = ("\nWhat environment variables do you want to forward?" <> showOld rtml_forwardEnvVars)
+  let readVars = map (dropWhileEnd isSpace . dropWhile isSpace) . splitOn ","
+  forwardEnvVars <- parseAnswer' rtml_forwardEnvVars readVars <$> askUserString alreadyExists True forwardEnvVarsPrompt "Env vars (comma separated list, e.g. VAR1,VAR2,VAR3): "
+
+  let newReachToml = ReachToml
+        { rtml_project = project
+        , rtml_cloudOrLocal = cloudOrLocal
+        , rtml_connector = connector
+        , rtml_forwardEnvVars = forwardEnvVars
+        }
+  path <- if alreadyExists then return oldTomlPath else getCurrentDirectory <&> (</> "reach.toml")
+  writeToml path reachTomlCodec newReachToml
+
+interactiveCreateGlobalsToml :: IO ()
+interactiveCreateGlobalsToml = do
+  oldToml <- readOrEmptyGlobalsToml
+  let showOld field = " Currently " <> maybe "unset" (\f -> "\"" <> show f <> "\"") (field oldToml)
+  let parseAnswer' field = parseAnswer $ field oldToml
+
```

Comment:
I hate extra newlines... I can't keep track of when the function ends

Reply:
Hm, I hate dense code for the same type of reason. My eyes and mind can't focus on it well. But if you want, I can remove the spacing.

Reply:
:cry: